File: common/src/main/java/juuxel/adorn/platform/EntityBridge.java
package juuxel.adorn.platform;

import dev.architectury.injectables.annotations.ExpectPlatform;
import juuxel.adorn.entity.SeatEntity;
import net.minecraft.entity.EntityType;
import org.jetbrains.annotations.NotNull;

public final class EntityBridge {
    @ExpectPlatform
    @NotNull
    public static EntityType<SeatEntity> createSeatType() {
        return PlatformCore.expected();
    }
}

File: common/src/main/kotlin/juuxel/adorn/platform/EntityBridge.kt
package juuxel.adorn.platform

import juuxel.adorn.entity.SeatEntity
import net.minecraft.entity.EntityType
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.util.math.BlockPos

interface EntityBridge {
    fun createSeatType(): EntityType<SeatEntity>
    fun PlayerEntity.trySleep(pos: BlockPos, onSuccess: () -> Unit)
}
